libtenzir/builtins/operators/show.cpp

```diff
@@ -0,0 +1,451 @@
+#include <tenzir/table_slice_builder.hpp>
+#include <tenzir/uuid.hpp>
+
+#include <caf/scoped_actor.hpp>
+#include <caf/timespan.hpp>
+
+#include <string>
+#include <vector>
+
+namespace tenzir::plugins::show {
+
+namespace {
+
+/// A type that represents a connector.
+auto connector_type() -> type {
+  return type{
+    "tenzir.connector",
+    record_type{
+      {"name", string_type{}},
+      {"loader", bool_type{}},
+      {"saver", bool_type{}},
+    },
+  };
+}
+
+/// A type that represents a format.
+auto format_type() -> type {
+  return type{
+    "tenzir.format",
+    record_type{
+      {"name", string_type{}},
+      {"printer", bool_type{}},
+      {"parser", bool_type{}},
+    },
+  };
+}
+
+/// A type that represents an operator.
+auto operator_type() -> type {
+  return type{
+    "tenzir.operator",
+    record_type{
+      {"name", string_type{}},
+      {"source", bool_type{}},
+      {"transformation", bool_type{}},
+      {"sink", bool_type{}},
+    },
+  };
+}
+
+/// A type that represents a partition.
+auto partition_type() -> type {
+  return type{
+    "tenzir.partition",
+    record_type{
+      {"uuid", string_type{}},
+      {"memory_usage", uint64_type{}},
+      {"min_import_time", time_type{}},
+      {"max_import_time", time_type{}},
+      {"version", uint64_type{}},
+      {"schema", string_type{}},
+    },
+  };
+}
+
+auto key_value_pair() -> record_type {
+  return record_type{
+    {"key", string_type{}},
+    {"value", string_type{}},
+  };
+}
+
+/// A type that represents a type attribute.
+auto type_attribute_type() -> type {
+  return type{
+    "tenzir.attribute",
+    key_value_pair(),
+  };
+}
+
+/// A type that represents a record field.
+auto record_field_type() -> type {
+  return type{
+    "tenzir.record_field",
+    record_type{
+      {"name", string_type{}},
+      {"type", string_type{}},
+    },
+  };
+}
+
+/// A type that represents a (Tenzir) type.
+/// The current record-based approach is a poorman's sum type approximation.
+/// With native union types, we'll be able to describe this more cleanly.
+auto type_type() -> type {
+  return type{
+    "tenzir.type",
+    record_type{
+      {"name", string_type{}},
+      {"structure", record_type{{"basic", string_type{}},
+                                {"enum", list_type{record_type{
+                                           {"name", string_type{}},
+                                           {"key", uint64_type{}},
+                                         }}},
+                                {"list", string_type{}},
+                                {"map", list_type{key_value_pair()}},
+                                {"record", list_type{record_field_type()}}}},
+      {"attributes", list_type{type_attribute_type()}},
+    },
+  };
+}
+
+/*
+/// Adds data to a field or list guard.
+void add(auto& guard, const data& x) {
+  auto f = detail::overload{
+    [&](const auto&) {
+      auto err = guard.add(make_view(x));
+      (void)err;
```

Comment:
Is it intended that this is left here?

Reply:
Not really. I thought I'll use it later on, but I now removed it with a dedicated commit so that it's easy to bring back.